tests/tdastro/astro_utils/test_passbands.py

```diff
@@ -0,0 +1,198 @@
+    with patch("numpy.loadtxt", side_effect=OSError("Mocked OSError")):
+        assert not passbands._load_transmission_table("test-band", "mock_file.dat")
+
+
+def test_passbands_load_all_transmission_tables(tmp_path):
+    """Test that we can download and load all transmission tables."""
+    band_ids = ["u", "g"]
+    passbands = Passbands(bands=band_ids)
+
+    # Mock the _get_data_path method
+    with patch.object(passbands, "_get_data_path"):
+        passbands.data_path = os.path.join(tmp_path, "band_data/")
+
+        # Mock the _download_transmission_table and _load_transmission_table methods
+        with patch.object(passbands, "_download_transmission_table", return_value=True), patch.object(
+            passbands, "_load_transmission_table", return_value=True
+        ):
+            passbands.load_all_transmission_tables()
+
+            # Assert _download_transmission_table was called for each band
+            assert passbands._download_transmission_table.call_count == len(band_ids)
+
+            # Assert _load_transmission_table was called for each band
+            assert passbands._load_transmission_table.call_count == len(band_ids)
+
+
+def test_passbands_phi_b():
+    """Test that we can calculate the value of phi_b for all wavelengths in a transmission table."""
+    passbands = Passbands()
+
+    u_band_transmission_table = np.array([[100.0, 0.5], [200.0, 0.75], [300.0, 0.25]])
+    g_band_transmission_table = np.array([[100.0, 0.75], [200.0, 0.5], [300.0, 0.25]])
+
+    u_band_phi_b = passbands._phi_b(u_band_transmission_table)
+    g_band_phi_b = passbands._phi_b(g_band_transmission_table)
+
+    assert len(u_band_phi_b) == len(u_band_transmission_table)
+    assert len(g_band_phi_b) == len(g_band_transmission_table)
+
+    # Check that the integrated value of u_band_phi_b about-equals that of g_band_phi_b
+    assert np.isclose(
+        np.trapz(u_band_phi_b, x=u_band_transmission_table[:, 0]),
+        np.trapz(g_band_phi_b, x=g_band_transmission_table[:, 0]),
+        rtol=1e-9,  # relative tolerance
+        atol=1e-9,  # absolute tolerance
+    )
+
+    # And that they're both about equal to 1.0
+    assert np.isclose(np.trapz(u_band_phi_b, x=u_band_transmission_table[:, 0]), 1.0, rtol=1e-9, atol=1e-9)
+    assert np.isclose(np.trapz(g_band_phi_b, x=g_band_transmission_table[:, 0]), 1.0, rtol=1e-9, atol=1e-9)
+
+
+def test_passbands_calculate_normalized_system_response_tables():
+    """Test that we can calculate the normalized system response tables for all bands."""
+    band_ids = ["u", "g"]
+    passbands = Passbands(bands=band_ids)
+
+    # Add mock transmission tables
+    passbands.transmission_tables["u"] = np.array([[100.0, 0.5], [200.0, 0.75], [300.0, 0.25]])
+    passbands.transmission_tables["g"] = np.array([[100.0, 0.75], [200.0, 0.5], [300.0, 0.25]])
+
+    # Mock the _phi_b method
+    with patch.object(passbands, "_phi_b", return_value=np.array([0.5, 0.5, 0.5])):
```

Comment:
Why override `_phi_b` here? Given there are only 6 entries in each of the two normalized_system_response_tables (and only checking the value of 3 of those), I think you could manually compute and check the expected answer instead of returning 0.5.

Reply:
calculate_normalized_system_response_tables is just calling _phi_b (tested just above) for all bands, so I wanted the focus to be on the new functionality introduced by this method--but you're right, it doesn't hurt to avoid mocking here.